prov/tcp/src/tcpx_progress.c

```diff
@@ -565,11 +562,7 @@ int tcpx_get_rx_entry_op_read_rsp(struct tcpx_ep *tcpx_ep)
 	rx_entry = container_of(entry, struct tcpx_xfer_entry,
 				entry);
 
-	memcpy(&rx_entry->hdr, &tcpx_ep->rx_detect.hdr,
-	       (size_t) tcpx_ep->rx_detect.hdr.base_hdr.payload_off);
-	rx_entry->hdr.base_hdr.op_data = TCPX_OP_READ_RSP;
-	rx_entry->rem_len = (rx_entry->hdr.base_hdr.size -
-			     tcpx_ep->rx_detect.done_len);
+	tcpx_get_rx_entry_op(tcpx_ep, rx_entry, TCPX_OP_READ_RSP);
```

Comment:
Of the 4 places where the code is updated, only 2 of them match.  Only 1 place sets mrecv_msg_start.  The other 3 set rem_len, which now isn't being set.  And only 3 set the ep.  I would drop this patch.

The function name is also confusing, as we're not 'getting' anything, only initializing some fields.

Reply:
@shefty Sorry for confusing, I dropped that patch.